### DISCUSSION TITLE
How to use relative URLs or specify a different web root for subdirectory setup?

### DISCUSSION THREAD

**Discussion by AlphaJack** [jonaswinkler/paperless-ng#893] [Feature Requests]
In my web pages I see hardcoded URLs like /static/ and /admin/. 

Can I host Paperless-ng in a subdirectory like /paperless/ and then have /paperless/static/ and /paperless/admin/? If yes, how?

**Comment by TheFehr** [comment]
Check https://github.com/jonaswinkler/paperless-ng/discussions/679 and https://github.com/jonaswinkler/paperless-ng/issues/36

**Reply by AlphaJack** [reply]
From https://example.org/paperless/ I get 301 redirected to https://example.org/accounts/login/?next=/paperless/, which doesn't exist, instead of https://example.org/paperless/accounts/login/?next=/paperless/.

Paperless-ng:
```ini
PAPERLESS_ALLOWED_HOSTS=example.org                                                                                               
PAPERLESS_CORS_ALLOWED_HOSTS=https://example.org/,http://127.0.0.1:8000/                                                          
PAPERLESS_FORCE_SCRIPT_NAME=/paperless                                                                                             
PAPERLESS_MEDIA_URL=/paperless/media/                                                                                              
PAPERLESS_STATIC_URL=/paperless/static/
```

NginX:
```nginx
location /paperless/ {
 proxy_pass http://127.0.0.1:8000/paperless/;
 proxy_http_version 1.1;
 proxy_redirect off;
 proxy_set_header Upgrade $http_upgrade;
 proxy_set_header Connection "upgrade";
 proxy_set_header Host $host;
 proxy_set_header X-Real-IP $remote_addr;
 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 proxy_set_header X-Forwarded-Host $server_name;
 proxy_set_header X-Script-Name /paperless;
}
```

I have also tried different `proxy_pass` values, but the result is the same.

Commenting out `PAPERLESS_FORCE_SCRIPT_NAME` instead redirects to the following page: https://example.org/accounts/login/?next=/

Other than /accounts, what about /admin and /ws? They seem to remain hardcoded to the root

**Reply by jonaswinkler** [reply]
See #36. What you're trying to do is impossible right now.
